[feat] 장바구니 관련 CRUD + 장바구니 제품 추천

AndBlack99 opened the pull request:
## 📌 PR 제목
[feat] 장바구니 관련 CRUD + 장바구니 제품 추천
---

## 📄 개요
- 장바구니 상품 조회
- 장바구니 상품 추가
- 장바구니 상품 수정
- 장바구니 상품 삭제    
- 장바구니 제품 추천

---

## ✅ 작업 내용
- [x] 기능 구현
- [ ] UI/UX 수정
- [x] API 연동
- [ ] 버그 수정
- [x] 테스트 코드 작성
- [x] 문서 작성

---

## 📃 작업 세부 내용

- Cart 전체 Repository 테스트
<img width="1350" alt="스크린샷 2025-07-03 오후 4 06 56" src="https://github.com/user-attachments/assets/7daf0df8-f30e-4552-bfc1-30b4aa2fc376" />

- 제품추천 ServiceImpl 테스트
<img width="1340" alt="스크린샷 2025-07-03 오후 4 07 31" src="https://github.com/user-attachments/assets/49610861-807d-4796-b73e-f6fcffa32817" />

- CartServiceImpl  테스트
<img width="1350" alt="스크린샷 2025-07-03 오후 4 07 48" src="https://github.com/user-attachments/assets/a37533bc-c703-4e12-9f18-0c5eb4de355b" />

---

## 🔍 관련 이슈
#34 

---

## 📝 참고 사항 
- AI 제품비교, S3 이미지 관련은 일단 제외하고 이후 새로운 이슈로 추가 하려고 합니다.
- 
---

## 🙏 리뷰 요청 사항

Commit:
feat : 초기구조 구축

Commit:
fix : UserPrincipal 사용자 식별 수정

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

Commit:
fix : Builder 어노테이션 추가

Commit:
feat : JpaConfig cart 경로 추가

Commit:
feat : swagger 어노테이션 추가

Commit:
fix : 최신 develop 반영

Commit:
fix : 최신 develop 반영

Commit:
fix : 최신 develop 삭제 미반영 파일 삭제

Commit:
test : 테스트코드 작성

Commit:
test : 중복 테스트코드 삭제

Commit:
refactor : 중복내용, 의미없는 주석 등 부분 수정

Commit:
feat : 장바구니 관련 ResponseCode 추가

Commit:
Fix: resolve merge conflict in JpaConfig.java

cake0420 commented on src/main/java/com/team5/catdogeats/carts/controller/CartController.java:
```diff
@@ -0,0 +1,128 @@
+            @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "401", description = "인증 실패"),
+            @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "404", description = "장바구니를 찾을 수 없음")
+    })
+    // 장바구니 목록 조회
```
하드코딩 말고 다른 방식은 없을까요

Commit:
fix: conflict해결

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

cake0420 commented on src/main/java/com/team5/catdogeats/auth/handler/OAuth2AuthenticationEntryPointHandler.java:
```diff
@@ -0,0 +1,30 @@
+package com.team5.catdogeats.auth.handler;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.AuthenticationEntryPoint;
+
+import java.io.IOException;
+
+public class OAuth2AuthenticationEntryPointHandler implements AuthenticationEntryPoint {
+    @Override
+    public void commence(HttpServletRequest request, HttpServletResponse response,
+                         AuthenticationException authException) throws IOException {
+
+        // 이미 인증된 사용자인지 확인
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth != null && auth.isAuthenticated() &&
+                !(auth instanceof AnonymousAuthenticationToken)) {
+            // 이미 인증된 사용자는 홈페이지로 리다이렉트
+            response.sendRedirect("/");
+            return;
+        }
+
+        // 미인증 사용자는 기본 동작 수행
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+    }
+}
```
이 파일은 제가 지웠던 걸로 기억하는데 깃허브 내역이 뭔가 꼬여있는지 들어가 있나보네여 지워주세요

AndBlack99 replied:
삭제했습니다!

cake0420 commented on src/main/java/com/team5/catdogeats/carts/service/impl/CartServiceImpl.java:
```diff
@@ -0,0 +1,241 @@
+@Transactional(readOnly = true)
+public class CartServiceImpl implements CartService {
+
+    private final CartRepository cartRepository;
```
빈 주입이 안되어있어서 커스텀 어노테이션으로 변경하시면 동작할겁니다

Commit:
remove : OAuth2AuthenticationEntryPointHandler.java 삭제

Commit:
refactor : 커스텀어노테이션 적용

Commit:
refactor: 장바구니 권한검증 로직 JOIN 기반 최적화

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

Commit:
feat: Cart 예외 처리 시스템 구축
- CartExceptionHandler 추가로 401/403/404/500 자동 처리
- Controller 코드 간소화 및 비즈니스 로직 집중

Commit:
refactor : 커스텀어노테이션 명칭 ApiResponse -> APIResponse 변경

Commit:
feat : 수량제한 시스템 추가
- 3단계 수량 제한 방어체계 구축 (엔티티/DTO/Service)
- 수량 범위: 1-10개 (0개 삭제 방지, 최대 10개 제한)

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

Commit:
feat: 장바구니 기반 상품 추천 기능 구현

Commit:
feat: 결제 완료 시 구매 상품 선별적 장바구니 삭제 시스템 구축
- OrderItemRepository 추가
- PaymentCompletedEvent 추가
- PaymentServiceImpl 수정(로직 추가)

Commit:
fix: JPQL 쿼리 Hibernate 6.x 호환성 오류 수정

- 복잡한 파라미터 조건을 Java 코드로 이동하여 해결

Commit:
fix: import 중복 제거

Commit:
fix : 테스트코드 수정

jungmyung16 commented on src/main/java/com/team5/catdogeats/orders/repository/OrderItemRepository.java:
```diff
@@ -0,0 +1,17 @@
+
+// 주문 상품 Repository
+// 결제 완료 시 구매한 상품 정보 조회를 위해 사용
+@Repository
```
repository 어노테이션 빼셔도 될거에요.

Commit:
fix : Repository 어노테이션 삭제

Commit:
test : 장바구니 제품추천 테스트코드 추가

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

Commit:
fix : 어노테이션명 변경 ApiResponse -> APIResponse

Commit:
fix : 하드코딩된 swagger 어노테이션 간소화

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/shopping-cart/34

Commit:
refactor : 제품추천 쿼리에서 직접 limit로 갯수 4개 제한으로 수정

Commit:
Merge

Commit:
test : 테스트코드 오류 해결

cake0420 commented:
수고하셨습니다

jungmyung16 reviewed (approved):
고생하셨습니다
